presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/BeginTableWrite.java

```diff
@@ -183,4 +205,30 @@ private PlanNode rewriteDeleteTableScan(PlanNode node, TableHandle handle, Rewri
             throw new IllegalArgumentException("Invalid descendant for DeleteNode: " + node.getClass().getName());
         }
     }
+
+    public static class Context
+    {
+        private Optional<WriterTarget> handle;
+        private Optional<WriterTarget> materializedHandle;
+
+        public Context()
+        {
+            this.handle = Optional.empty();
+            this.materializedHandle = Optional.empty();
+        }
+
+        public void addMaterializedHandle(WriterTarget handle, WriterTarget materializedHandle)
+        {
+            this.handle = Optional.of(handle);
+            this.materializedHandle = Optional.of(materializedHandle);
+        }
+
+        public Optional<WriterTarget> getMaterializedHandle(WriterTarget handle)
+        {
+            if (this.handle.isPresent()) {
+                checkState(this.handle.get().equals(handle), "can only have one WriterTarget in a subtree");
```

Comment:
You can write this as `ifPresent`. But in this case, I believe the `if` should be removed.